exchanges/gateio/gateio_websocket.go

```diff
@@ -31,77 +31,86 @@ const (
 // WsConnect initiates a websocket connection
 func (g *Gateio) WsConnect() error {
 	if !g.Websocket.IsEnabled() || !g.IsEnabled() {
-		return errors.New(wshandler.WebsocketNotEnabled)
+		return errors.New(stream.WebsocketNotEnabled)
 	}
 	var dialer websocket.Dialer
-	err := g.WebsocketConn.Dial(&dialer, http.Header{})
+	err := g.Websocket.Conn.Dial(&dialer, http.Header{})
 	if err != nil {
 		return err
 	}
 	go g.wsReadData()
-	_, err = g.wsServerSignIn()
+
+	if g.Websocket.CanUseAuthenticatedEndpoints() {
+		err = g.wsServerSignIn()
+		if err != nil {
+			return err
```

Comment:
Thanks for changing it with the server sign in, but I'm still on the view that a failure of auth should not impact the main websocket and am recommending it be changed to 
```
if g.GetAuthenticatedAPISupport(exchange.WebsocketAuthentication) {
		err = g.wsServerSignIn()
		if err != nil {
			g.Websocket.DataHandler <- err
			g.Websocket.SetCanUseAuthenticatedEndpoints(false)
		} else {
			var authsubs []stream.ChannelSubscription
			authsubs, err = g.GenerateAuthenticatedSubscriptions()
			if err != nil {
				g.Websocket.DataHandler <- err
				g.Websocket.SetCanUseAuthenticatedEndpoints(false)
			}
			err = g.Websocket.SubscribeToChannels(authsubs)
			if err != nil {
				g.Websocket.DataHandler <- err
				g.Websocket.SetCanUseAuthenticatedEndpoints(false)
			}
		}
	}
```
I'm open to considering otherwise, like I understand that if there's something wrong with auth sub generation its pretty bad, but it prevents the normal websocket from working

Reply:
Always a critic! 🤭 , Yeah will change just to be on the safe side.